csunplugged/utils/group_lessons_by_age.py

```diff
@@ -16,14 +16,15 @@ def group_lessons_by_age(lessons):
         A ordered dictionary of grouped lessons.
         The key is a tuple of the minimum age and maximum ages for
         the lessons.
-        The value for a key is a sorted list of lessons.
-        The dictionary is ordered by minimum age, then maximum age.
+        The value for a key is a sorted list of lessons (ordered by number).
     """
     grouped_lessons = OrderedDict()
-    lessons = lessons.order_by("min_age", "max_age", "number")
+    lessons = lessons.order_by("number")
     for lesson in lessons:
-        if (lesson.min_age, lesson.max_age) in grouped_lessons:
-            grouped_lessons[(lesson.min_age, lesson.max_age)].append(lesson)
-        else:
-            grouped_lessons[(lesson.min_age, lesson.max_age)] = [lesson]
+        for age_range in lesson.age_range.all():
+            ages = (age_range.age_range.lower, age_range.age_range.upper)
+            if ages in grouped_lessons.keys():
+                grouped_lessons[ages].append(lesson)
+            else:
+                grouped_lessons[ages] = [lesson]
     return grouped_lessons
```

Comment:
I have a feeling this won't guarantee the order of lessons and age groups as we always expect it, however this will be solved in #437.